tests/unit/Rules/AbstractSearcherTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of Respect/Validation.
+ *
+ * (c) Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+/**
+ * @group  rule
+ * @covers Respect\Validation\Rules\AbstractSearcher
+ */ 
+
+class AbstractSearcherTest extends \PHPUnit_Framework_TestCase
+{
+
+    public function testValidateMethodWhenArgsIsNull()
+    {
+        $compositeRuleMock = $this->getMockForAbstractClass('Respect\\Validation\\Rules\\AbstractSearcher');
+        $compositeRuleMock->expects($this->any())
+             ->method('validate')
```

Comment:
You shouldn't have mocked the `validate` method as it's the concrete method we want to test. You only need to mock the abstract class (as you did), pass an array to the `$haystack` attribute and do some tests against the conditionals in the `validate` method.

Through the `validate` method, you'll be able to cover `validateEquals` and `validateIdentical`.